subworkflows/local/markdup_stats.nf

```diff
@@ -0,0 +1,60 @@
+//
+// Based on https://github.com/sanger-tol/readmapping/blob/5a2657f5274a2c436b3896497a1322f2557bd9a8/subworkflows/local/markdup_stats.nf
+// from Sanger readmapping pipeline by @priyanka-surana
+//
+// Merge and Markdup all alignments at specimen level
+// Convert to CRAM and calculate statistics
+//
+include { SAMTOOLS_SORT } from '../../modules/nf-core/samtools/sort/main'
+include { SAMTOOLS_VIEW as SAMTOOLS_VIEW_MARKDUP } from '../../modules/nf-core/samtools/view/main'
+include { MARKDUPLICATE } from '../../subworkflows/local/markduplicate'
+include { CONVERT_STATS } from '../../subworkflows/local/convert_stats'
+include { BEDTOOLS_BAMTOBED } from '../../modules/nf-core/bedtools/bamtobed/main'
+include { BEDTOOLS_SORT } from '../../modules/nf-core/bedtools/sort/main'
+
+workflow MARKDUP_STATS {
+    take:
+    aln // channel: [ val(meta), [ sam ] ]
+    fasta // channel: /path/to/fasta
+
+    main:
+    ch_versions = Channel.empty()
+
+    // Sort SAM and convert to BAM
+    SAMTOOLS_SORT ( aln )
+    ch_versions = ch_versions.mix(SAMTOOLS_SORT.out.versions.first())
+
+    // Collect all BWAMEM2 output by sample name
+    SAMTOOLS_SORT.out.bam
+    .groupTuple(by: [0])
+    .set { ch_bams }
+
+    // Mark duplicates
+    MARKDUPLICATE ( ch_bams )
+    ch_versions = ch_versions.mix(MARKDUPLICATE.out.versions)
+
+    // Convert merged BAM to CRAM and calculate indices and statistics
+    ch_stat = MARKDUPLICATE.out.bam.map { meta, bam -> [ meta, bam, [] ] }
+  
+    SAMTOOLS_VIEW_MARKDUP ( ch_stat, fasta, [] )
+    ch_versions = ch_versions.mix(SAMTOOLS_VIEW_MARKDUP.out.versions)
+
+    BEDTOOLS_BAMTOBED( SAMTOOLS_VIEW_MARKDUP.out.bam )
```

Comment:
`bedtools_bamtobed` already has bedtools `sort` but `bedtools_sort` is not very efficient. See disclaimer at the bottom: https://bedtools.readthedocs.io/en/latest/content/tools/sort.html